sdv/data_processing/data_processor.py

```diff
@@ -323,138 +323,433 @@ def create_regex_generator(self, column_name, sdtype, column_metadata, is_numeri
+        """Get a datetime transformer.
+
+        Args:
+            kwargs (dict):
+                The keyword arguments for the transformer.
+            transformer (rdt.transformers.DatetimeFormatter):
+                A datetime transformer.
+
+        Returns:
+            rdt.transformers.DatetimeFormatter:
+                A datetime transformer.
+        """
+        kwargs['enforce_min_max_values'] = self._enforce_min_max_values
+
+        return self._get_other_transformer(kwargs, transformer)
+
+    def _get_categorical_transformer(self, kwargs, transformer):
+        """Get a categorical transformer.
+
+        Args:
+            kwargs (dict):
+                The keyword arguments for the transformer.
+            transformer:
+                A categorical transformer.
+
+        Returns:
+            transformer:
+                A categorical transformer.
+        """
+        return self._get_other_transformer(kwargs, transformer)
+
+    def _get_numerical_transformer(self, kwargs, transformer):
+        """Get a numerical transformer.
+
+        Args:
+            kwargs (dict):
+                The keyword arguments for the transformer.
+            transformer:
+                A numerical transformer.
+
+        Returns:
+            transformer:
+                A numerical transformer.
+        """
+        return self._get_other_transformer(kwargs, transformer)
+
+    def _get_boolean_transformer(self, kwargs, transformer):
+        """Get a boolean transformer.
+
+        Args:
+            kwargs (dict):
+                The keyword arguments for the transformer.
+            transformer:
+                A boolean transformer.
+
+        Returns:
+            transformer:
+                A boolean transformer.
+        """
+        return self._get_other_transformer(kwargs, transformer)
+
+    def _get_sdtype_transformer(self, sdtype, kwargs, transformer):
+        transformer_map = {
+            'datetime': self._get_datetime_transformer,
+            'numerical': self._get_numerical_transformer,
+            'categorical': self._get_categorical_transformer,
+            'boolean': self._get_boolean_transformer,
+        }
+        transformer_fn = transformer_map.get(sdtype)
+        if transformer_fn:
+            return transformer_fn(kwargs, transformer)
+
+        return self._get_other_transformer(kwargs, transformer)
+
     def _get_transformer_instance(self, sdtype, column_metadata):
         transformer = self._transformers_by_sdtype[sdtype]
-        if isinstance(transformer, AnonymizedFaker):
-            is_lexify = transformer.function_name == 'lexify'
-            is_baseprovider = transformer.provider_name == 'BaseProvider'
-            if is_lexify and is_baseprovider:  # Default settings
-                return self.create_anonymized_transformer(
-                    sdtype, column_metadata, None, self._locales
-                )
+        if transformer is None:
+            return None
```

Comment:
This is for when we set the categorical transformer to `None` in CTGAN ?

Reply:
I'm not sure. This logic was already in the code so I kept it.